src/oak_d.py

```diff
@@ -397,10 +399,20 @@ async def get_images(
         # Accumulator for timestamp calculation later
         seconds_float: float = None
 
-        if COLOR_SENSOR in self.sensors:
-            captured_data = await cls.worker.get_color_image()
-            arr, captured_at = captured_data.np_array, captured_data.captured_at
+        # We really want to do this as in-sync as possible to reduce latency between color & depth
+        color_data: Optional(CapturedData) = None
+        depth_data: Optional(CapturedData) = None
+        if COLOR_SENSOR in self.sensors and DEPTH_SENSOR in self.sensors:
+            color_data, depth_data = await asyncio.gather(
+                cls.worker.get_color_image(), cls.worker.get_depth_map()
```

Comment:
Do we need any error handling around image fetches?

Reply:
Hmm, maybe we should implement a timeout or something because right now there are no errors because we would perpetually hang if no image fetch happens. See the while loop in `get_color_image` and `get_depth_map`

Reply:
ok good to know. probably not necessary to complicate with timeout right now